Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -1565,17 +1565,21 @@ public static InputDevice GetDevice(string nameOrLayout)
         public static TDevice GetDevice<TDevice>()
             where TDevice : InputDevice
         {
-            TDevice result = null;
+            return (TDevice)GetDevice(typeof(TDevice));
+        }
+
+        public static InputDevice GetDevice(Type type)
+        {
+            InputDevice result = null;
             var lastUpdateTime = -1.0;
             foreach (var device in devices)
             {
-                var deviceOfType = device as TDevice;
-                if (deviceOfType == null)
+                if (device.GetType() != type)
```

Comment:
Should use `IsAssignableFrom` to take inheritance into account.

Reply:
Fixed